packages/api/src/rich-text/util.ts

```diff
@@ -0,0 +1,6 @@
+export const MENTION_REGEX = /(^|\s|\()(@)([a-zA-Z0-9.-]+)(\b)/g
+export const URL_REGEX =
+  /(^|\s|\()((https?:\/\/[\S]+)|((?<domain>[a-z][a-z0-9]*(\.[a-z0-9]+)+)[\S]*))/gim
+export const TAG_REGEX =
+  /(^|\s)[#＃]((?!\ufe0f|\u20e2)[^\s\u00AD\u2060\u200A\u200B\u200C\u200D\u20e2]*[^\p{P}\d\s\u00AD\u2060\u200A\u200B\u200C\u200D\u20e2]+[^\s\u00AD\u2060\u200A\u200B\u200C\u200D\u20e2]*)?/gu
```

Comment:
the punctuation character set includes zwj zws and the other zero width characters 

now that it's like this, the core matching could be simplified to `(?!\ufe0f|\u20e3)[\d]*[^ \n\d\p{P}]+[\d]*` i suppose?

Reply:
Testing, but appears that `/\p{P}/u.test(<zws>)` is `false` 🤔

I'd love to try to capture them all, or a range, but this regex will likely continue to grow either way

Reply:
Looks like `Cf` works, wonder if that breaks anything...

Reply:
`\p{Cf}\p{Me}` 🤔

Reply:
Matches some characters we might as well support though, like arabic number sign